src/DynamoCore/Scheduler/UpdateGraphAsyncTask.cs

```diff
@@ -63,6 +63,8 @@ internal bool Initialize(EngineController controller, WorkspaceModel workspace)
                 if (graphSyncData == null)
                     return false;
 
+                engineController.UpdateProfilingData(workspace.Nodes);
```

Comment:
is there a more specific location this can be handled in - the profiling session itself or is this the cleanest?

Reply:
Moved to the profiling session, engine controller methods cleaned up and test updated to reflect the change.